### PR TITLE
Getting minimal InteractionController functionality

### PR DESCRIPTION
**Pull request by eozaki**
- [x] Gradle adequation to Kotlin format;
 - [x] Gradle dependencies;
 - [x] Other Gradle miscellania;
 - [x] Karaoke application class;
 - [x] InteractionController with a POST route for new interaction;
 - [x] Automated tests;

@Talkdesk/routing-platform aberta a temporada de review!

### CODE REVIEW

File: src/main/kotlin/com/karaoke/orchestrator/InteractionController.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import java.util.concurrent.atomic.AtomicLong
+
+@RestController
+class InteractionController {
+    val counter = AtomicLong()
+
+    @PostMapping("/interaction")
```
[the remark this answers is not in the document]
Comment (eozaki) [reply src/main/kotlin/com/karaoke/orchestrator/InteractionController.kt]: My mistake on the plural thing, for a fact.

When it comes to the `@RequestMapping`, I saw that property on the tutorials, but the challenge states:
```3. On the work branch create a spring boot app with Kotlin with an Orchestrator. It should expose an HTTP POST endpoint that will later be used to run Interactions by receiving a routing configuration ID. At the moment, it should just log the received ID.```
So, I believe, no need for other routes, at least as of right now.

File: src/test/kotlin/com/karaoke/orchestrator/InteractionControllerTest.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+@AutoConfigureMockMvc
+class InteractionControllerTest() {
+    @Autowired
+    private lateinit var mockController: MockMvc
+
+    @Test
+    fun testMinimalPost() {
+        val body = """
+              {
+                  "receiverId": "+351910986347"
```
[the remark this answers is not in the document]
Comment (eozaki) [reply src/test/kotlin/com/karaoke/orchestrator/InteractionControllerTest.kt]: Both my number, and my bad not to remember taking it out.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.gradle/
+build/**/*
+.idea/**/*
```
[the remark this answers is not in the document]
Comment (eozaki) [reply .gitignore]: So am I
Still learning how to deal with the Kotlin/intelliJ thing
Any suggestions on how to deal with the ignored files? Any pattern on which ones we should usually overlook and which ones we should commit? Company recomendations?

[the remark this answers is not in the document]
Comment (sergiofbsilva) [reply .gitignore]: crazy stuff indeed :)
try to use this one and see if it works https://github.com/Talkdesk/orchestrator/blob/master/.gitignore

I guess the problem is that you have an initial commit with all the files already there :/

File: src/main/kotlin/com/karaoke/orchestrator/InteractionController.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (sergiofbsilva) [reply src/main/kotlin/com/karaoke/orchestrator/InteractionController.kt]: There are no mistakes we are constantly learning or remembering stuff we've learnt before 😎 
Maybe you can tackle my comment later on as the onboarding challenge evolves 👍

**Comment by eozaki** [comment]
> Well done! Congrats on your first PR 🎉
> 
> I suggest moving this onboarding challenge to a private repo of yours due to our limited restrictions on repos and github limits. If you need help with this let me know.
> Can you try to squash your commits so that the changeset is concise and clear ?
> 
> Make sure to follow the [guidelines on writing good commit messages](https://github.com/erlang/otp/wiki/writing-good-commit-messages)

The challenge description suggests a private repo on Talkdesk's workspace. I tried to get one of those, but could not since company private repo's have to be paid for. So, I had to decide private repo of my own and include everyone by hand, while having the chance of forgetting someone. Or public repo in here with everyone included. Even talked to @joao-espada-santos-talkdesk about it back then. I can move it all after the PR is approved or closed.

File: .gitignore (hunk 3 above)
[the remark this answers is not in the document]
Comment (eozaki) [reply .gitignore]: Done, @sergiofbsilva

[the remark this answers is not in the document]
Comment (eozaki) [reply .gitignore]: Done, @sergiofbsilva

**Pull request merged**